api/lib/domain/services/schooling-registrations-xml-service.js

```diff
@@ -1,91 +1,210 @@
+          try {
+            if (err) throw err;
+            processStudentsNodes(mapSchoolingRegistrationsByStudentId, nodeData, nationalStudentIds);
+            processStudentsStructureNodes(mapSchoolingRegistrationsByStudentId, nodeData);
+          } catch (err) {
+            reject(err);
+          }
+        });
+      }
+    });
+
+    siecleFileStream.pipe(saxParser);
+
+    streamerToParseSchoolingRegistrations.on('end', () => {
+      resolve(Array.from(mapSchoolingRegistrationsByStudentId.values()));
+    });
+  });
 }
 
-function _filterSchoolingRegistrationsWithNoDivision(xmlSchoolingRegistration, xmlSchoolingRegistrationsStructures) {
-  const schoolingRegistrationStructuresContainer = _findSchoolingRegistrationStructures(xmlSchoolingRegistration, xmlSchoolingRegistrationsStructures);
-  const schoolingRegistrationDivision = _findSchoolingRegistrationDivision(schoolingRegistrationStructuresContainer);
-  return !_.isEmpty(schoolingRegistrationDivision) && schoolingRegistrationDivision !== 'Inactifs';
+function _mapStudentInformationToSchoolingRegistration(nodeData) {
+  return {
+    lastName: _getValueFromParsedElement(nodeData.ELEVE.NOM_DE_FAMILLE),
+    preferredLastName: _getValueFromParsedElement(nodeData.ELEVE.NOM_USAGE),
+    firstName: _getValueFromParsedElement(nodeData.ELEVE.PRENOM),
+    middleName: _getValueFromParsedElement(nodeData.ELEVE.PRENOM2),
+    thirdName: _getValueFromParsedElement(nodeData.ELEVE.PRENOM3),
+    birthdate: moment(nodeData.ELEVE.DATE_NAISS, 'DD/MM/YYYY').format('YYYY-MM-DD') || null,
+    birthCountryCode: _getValueFromParsedElement(nodeData.ELEVE.CODE_PAYS, null),
+    birthProvinceCode: _getValueFromParsedElement(nodeData.ELEVE.CODE_DEPARTEMENT_NAISS),
+    birthCityCode: _getValueFromParsedElement(nodeData.ELEVE.CODE_COMMUNE_INSEE_NAISS),
+    birthCity: _getValueFromParsedElement(nodeData.ELEVE.VILLE_NAISS),
+    MEFCode: _getValueFromParsedElement(nodeData.ELEVE.CODE_MEF),
+    status: _getValueFromParsedElement(nodeData.ELEVE.CODE_STATUT),
+    nationalStudentId: _getValueFromParsedElement(nodeData.ELEVE.ID_NATIONAL),
+  };
 }
 
-function _filterNotYetArrivedSchoolingRegistrations(xmlSchoolingRegistration) {
-  const nationalSchoolingRegistrationId = _getEitherElementValueOrNull(xmlSchoolingRegistration, 'ID_NATIONAL');
-  return !_.isEmpty(nationalSchoolingRegistrationId);
+async function _readFirstLineFromFile(path) {
+  return new Promise((resolve, reject) => {
+    const readStream = fs.createReadStream(path);
+    const lineEnding = '\n';
+    const BOM = 0xFEFF;
+    let value = '';
+    let position = 0;
+    let index;
+    readStream.on('data', (chunk) => {
+      index = chunk.indexOf(lineEnding);
+      value += chunk;
+      if (index === -1) {
+        position += chunk.length;
+      } else {
+        position += index;
+        readStream.close();
+      }
+    })
+      .on('close', () => resolve(value.slice(value.charCodeAt(0) === BOM ? 1 : 0, position)))
```

Comment:
Si on voulait être TRES explicite pour ceux qui ne savent pas que le premier caractère de début de fichier peut ne pas être significatif sur le contenu de la ligne ( et indique uniquement le format)

```suggestion
    const returnExtractedFirstLine = function() {
      const rawFirstLine = value;
      const lineStartsAt = rawFirstLine.charCodeAt(0) === BOM ? 1 : 0;
      const lineEndsAt = position;
      const firstLine = rawFirstLine.slice(lineStartsAt, lineEndsAt);
      resolve(firstLine);
    };
    // (...)
          .on('close', returnExtractedFirstLine)   
```

Reply:
D'ailleurs ici pour culture info, c'est quel cas sur les 3 ?
-     The byte order, or endianness, of the text stream in the cases of 16-bit and 32-bit encodings;
-     The fact that the text stream's encoding is Unicode, to a high level of confidence;
-     Which Unicode character encoding is used.

Si les fichiers SIECLE sont en ISO-8859-15, je penche pour `Which Unicode character encoding is used.`
https://en.wikipedia.org/wiki/Byte_order_mark